data-runtime/src/main/java/io/micronaut/data/runtime/mapper/QueryStatement.java

```diff
@@ -100,7 +100,7 @@ default QueryStatement<PS, IDX> setDynamic(
                 if (value instanceof Date date) {
                     return setDate(statement, index, date);
                 } else {
-                    return setDate(statement, index, convertRequired(value, Date.class));
+                    return setDate(statement, index, convertRequired(value, java.sql.Date.class));
```

Comment:
Need to convert to java.sql.Date or else converter from core that converts to UTC will be used. And we need sql date anyway.